hummingbot/core/utils/trading_pair_fetcher.py

```diff
@@ -49,14 +51,22 @@ async def fetch_all(self, client_config_map: ClientConfigAdapter):
                         connector_setting=connector_settings[conn_setting.parent_name],
                         connector_name=conn_setting.name
                     )
+                elif not self.fetch_pairs_from_all_exchanges:
```

Comment:
I made a change to this simplifying the logic to check only if file exist and moved the logic into all_connector_settings

Reply:
hi @isreallee82, the changes that are requested are different, this is almost the same as you have before. Instead of adding the if-else, just modify the method _all_connector_settings

Reply:
sorry about that I couldn't get better understanding of the "abstracting the logic" Thanks @cardosofede this method is seem better

Reply:
hi @cardosofede I noticed using that logic I couldn't fetch paper trades exchanges just connected + connected_paper_tades